### PR TITLE
Sonar fixes

### PR DESCRIPTION
**Pull request by robertkrantz**

### CODE REVIEW

File: service/src/test/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FishingActivityDaoTest.java [hunk 1]
```diff
@@ -50,38 +48,13 @@ public void prepare() {
     @Test
     @SneakyThrows
     public void findEntityById() {
-        // Given
```
Comment (RobinMilding) [inline service/src/test/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FishingActivityDaoTest.java:53]: I don't know, I kind of like having all three comments present, even when they don't "contain" anything...

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FaReportDocumentDao.java [hunk 2]
```diff
@@ -172,18 +163,19 @@ public FaReportDocumentEntity findFaReportByRefIdAndRefScheme(String reportRefId
         for (FaReportDocumentEntity report : reports) {
             populateDeletingAndCancellationIds(report.getFishingActivities(), idsOfCancelledDeletedReports);
         }
+
         if (CollectionUtils.isNotEmpty(idsOfCancelledDeletedReports)) {
             return findReportsByIdsList(idsOfCancelledDeletedReports);
         }
-        return null;
+        return new ArrayList<>();
```
Comment (RobinMilding) [inline service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FaReportDocumentDao.java:170]: Are you sure that no-one is checking to see if the returned value was null? I feel like there's a lot of that sort of code around... (also I'm getting déjà vu, I'm pretty sure I've written something exactly like this before in a review comment...)

**Review by RobinMilding — APPROVED** [review]
Good stuff!

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FaReportDocumentDao.java (hunk 2 above)
Comment (robertkrantz) [reply service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FaReportDocumentDao.java:170]: Only place that uses it does an empty check

**Pull request merged**